app/src/main/java/com/github/polypoly/app/ui/game/GameEndUIComponents.kt

```diff
@@ -0,0 +1,142 @@
+package com.github.polypoly.app.ui.game
+
+import android.content.Intent
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.window.Dialog
+import com.github.polypoly.app.base.game.Game
+import com.github.polypoly.app.base.game.Player
+import com.github.polypoly.app.ui.theme.Padding
+import com.github.polypoly.app.ui.theme.PolypolyTheme
+import com.github.polypoly.app.ui.theme.UIElements
+
+/**
+ * All the UI displayed at the end of a game is available here
+ */
+
+// ============================================================== GLOBAL UI
+
+@Composable
+fun GameEndUI(lastGame: Game, quitAction: () -> Unit) {
+    Dialog(onDismissRequest = {}) {
+        Surface(
+            modifier = Modifier
+                .padding(Padding.medium)
+                .testTag("end_screen"),
+            color = Color.Transparent
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxWidth(0.9f)
+                    .fillMaxHeight(0.5f),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center
+            ) {
+                GameBoard(getPlayersInRankingOrder(lastGame))
+                Spacer(modifier = Modifier.height(Padding.large))
+                UIElements.BigButton(onClick = quitAction, text = "Return to menu")
+            }
+
+        }
+    }
+}
+
+// ============================================================== PRIVATE COMPONENTS
+
+@Composable
+private fun GameBoard(orderedPlayers: List<Player>) {
+    Surface(
+        shape = RoundedCornerShape(Padding.medium)
+    ) {
+        Column(
+            modifier = Modifier
+                .background(MaterialTheme.colors.primary)
+                .padding(Padding.large),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(Padding.large)
+        ) {
+            Text(
+                text = "End of the game!",
```

Comment:
This one can also be extracted as a string ressource.

Reply:
same as above